### ISSUE TITLE
Installation fails on fresh Ubuntu 20.04 on Vultr

### ISSUE THREAD

**Issue by prologic** [IceWhaleTech/CasaOS#429]
Following the instructions on the [README](/README.md) the installation fails on a fresh Ubuntu 20.04 VM running on Vultr.

This is what I called a "README Fail".

See:

```#!sh

   _____                 ____   _____
  / ____|               / __ \ / ____|
 | |     __ _ ___  __ _| |  | | (___
 | |    / _` / __|/ _` | |  | |\___ \
 | |___| (_| \__ \ (_| | |__| |____) |
  \_____\__,_|___/\__,_|\____/|_____/

   --- Made by IceWhale with YOU ---

[  OK  ] Your hardware architecture is : x86_64
[  OK  ] Your System is : Linux
[  OK  ] Your Linux Distribution is : ubuntu
[  OK  ] Memory capacity check passed.
[  OK  ] Disk capacity check passed.
[ INFO ] Install the necessary dependencies: smartmontools
Reading package lists...
Building dependency tree...
Reading state information...
Suggested packages:
  gsmartcontrol smart-notifier mailx | mailutils
The following NEW packages will be installed:
  smartmontools
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 525 kB of archives.
After this operation, 2,087 kB of additional disk space will be used.
Get:1 http://au.clouds.archive.ubuntu.com/ubuntu focal/main amd64 smartmontools amd64 7.1-1build1 [525 kB]
Fetched 525 kB in 3s (183 kB/s)
Selecting previously unselected package smartmontools.
(Reading database ... 118761 files and directories currently installed.)
Preparing to unpack .../smartmontools_7.1-1build1_amd64.deb ...
Unpacking smartmontools (7.1-1build1) ...
Setting up smartmontools (7.1-1build1) ...
Created symlink /etc/systemd/system/smartd.service → /lib/systemd/system/smartmontools.service.
Created symlink /etc/systemd/system/multi-user.target.wants/smartmontools.service → /lib/systemd/system/smartmontools.service.
Processing triggers for man-db (2.9.1-1) ...
Processing triggers for systemd (245.4-4ubuntu3.17) ...
[ INFO ] Install the necessary dependencies: net-tools
Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  net-tools
0 upgraded, 1 newly installed, 0 to remove and 0 not upgraded.
Need to get 196 kB of archives.
After this operation, 864 kB of additional disk space will be used.
Get:1 http://au.clouds.archive.ubuntu.com/ubuntu focal/main amd64 net-tools amd64 1.60+git20180626.aebd88e-1ubuntu1 [196 kB]
Fetched 196 kB in 2s (112 kB/s)
Selecting previously unselected package net-tools.
(Reading database ... 118806 files and directories currently installed.)
Preparing to unpack .../net-tools_1.60+git20180626.aebd88e-1ubuntu1_amd64.deb ...
Unpacking net-tools (1.60+git20180626.aebd88e-1ubuntu1) ...
Setting up net-tools (1.60+git20180626.aebd88e-1ubuntu1) ...
Processing triggers for man-db (2.9.1-1) ...
[  OK  ] Docker will be installed automatically.
# Executing docker install script, commit: b2e29ef7a9a89840d2333637f7d1900a83e7153f
+ sh -c 'apt-get update -qq >/dev/null'
E: Repository 'https://apprepo.vultr.com/ubuntu universal InRelease' changed its 'Codename' value from 'buster' to 'universal'
[  OK  ] Docker Successfully installed.
[FAILED] Docker is not running, try to start
Failed to start docker.service: Unit docker.service not found.
[FAILED] Docker is not running, try to start
Failed to start docker.service: Unit docker.service not found.
[FAILED] Docker is not running, try to start
Failed to start docker.service: Unit docker.service not found.
[ INFO ] Create folders.
[ INFO ] Start cleaning up the old version.
[  OK  ] Clearance completed.
[ INFO ] Downloading CasaOS for linux/amd64...
[  OK  ] Download successful!
[ INFO ] Extracting...
[ INFO ] Installing CasaOS Addons
[ INFO ] Installing...
[  OK  ] CasaOS Successfully installed.
[ INFO ] Create system service for CasaOS.
[  OK  ] CasaOS service Successfully created.
[ INFO ] Create a system startup service for CasaOS.
Created symlink /etc/systemd/system/multi-user.target.wants/casaos.service → /etc/systemd/system/casaos.service.
[ INFO ] Start CasaOS service.
[  OK  ] Service started successfully.
[  OK  ] Clean...
 ─────────────────────────────────────────────────────
 CasaOS v0.3.4 is running at:
 ─────────────────────────────────────────────────────
 - http://xxx.xxx.xx.xxx (enp1s0)
 Open your browser and visit the above address.
 ─────────────────────────────────────────────────────

 CasaOS Project  : https://github.com/IceWhaleTech/CasaOS
 CasaOS Team     : https://github.com/IceWhaleTech/CasaOS#maintainers
 CasaOS Discord  : https://discord.gg/knqAbbBbeX
 Website         : https://www.casaos.io
 Online Demo     : http://demo.casaos.io

 Uninstall       : casaos-uninstall

root@cloudron:~# docker ps

Command 'docker' not found, but can be installed with:

snap install docker     # version 20.10.14, or
apt  install docker.io  # version 20.10.12-0ubuntu2~20.04.1

See 'snap info docker' for additional versions.

root@cloudron:~#
root@cloudron:~# logout
Connection to xxx.xxx.xx.xxx closed.
```

Please ignore the `cloudron` hostname 😅 I've been testing Cloudron (FAILED) and Umbrel (INSECURE) to 😂

**Comment by JohnGuan** [comment]
Try uninstalling docker and reinstalling it separately, then install CasaOS.

**Comment by prologic** [comment]
Isn't CasaOS suppose to install Docker automatically?

**Comment by JohnGuan** [comment]
> Isn't CasaOS suppose to install Docker automatically?

@prologic Yes, that's right. But at the moment it looks like the installation script is having problems. We've traced the issue in #392

**Comment by jerrykuku** [comment]
CasaOS uses the official Docker installation method to install Docker, and we have a small issue with the status detection of docker. We will try to fix this issue as soon as possible.

**Comment by kobykotiv** [comment]
I have a similar issue...

I tried installing casaos on a vultr instance. Debian 11, Ubuntu 20.04 LTS, and vultr's "one-click-docker" installation. I cannot connect to casa via `MYSERVERIP:80` SSH works fine.

When I prod my server with `sudo lsof -i -P -n | grep LISTEN` 

`systemd-r  580 systemd-resolve   13u  IPv4  21084      0t0  TCP 127.0.0.53:53 (LISTEN)
casaos     599            root    8u  IPv6  25020      0t0  TCP *:80 (LISTEN)
casaos     599            root    9u  IPv6  25023      0t0  TCP *:36309 (LISTEN)
sshd       660            root    3u  IPv4  22411      0t0  TCP *:22 (LISTEN)
sshd       660            root    4u  IPv6  22422      0t0  TCP *:22 (LISTEN)`

Oddly there is no IPv4 listening port

I tried running `sudo ss -tulpn | grep LISTEN` and got

`tcp    LISTEN  0       4096             127.0.0.53%lo:53           0.0.0.0:*     users:(("systemd-resolve",pid=580,fd=13))                                      
tcp    LISTEN  0       128                    0.0.0.0:22           0.0.0.0:*     users:(("sshd",pid=660,fd=3))                                                  
tcp    LISTEN  0       4096                         *:80                 *:*     users:(("casaos",pid=599,fd=8))                                                
tcp    LISTEN  0       4096                         *:36309              *:*     users:(("casaos",pid=599,fd=9))                                                
tcp    LISTEN  0       128                       [::]:22              [::]:*     users:(("sshd",pid=660,fd=4))`


I have a working casa instance running on a Digital Ocean Droplet and I can connect to it via `MYDOSERVER:80` just fine. It runs Debian 11, 

`tcp   LISTEN 0      128          0.0.0.0:22         0.0.0.0:*    users:(("sshd",pid=621,fd=3))
tcp   LISTEN 0      4096         0.0.0.0:8088       0.0.0.0:*    users:(("docker-proxy",pid=132822,fd=4))
tcp   LISTEN 0      4096         0.0.0.0:443        0.0.0.0:*    users:(("docker-proxy",pid=132842,fd=4))
tcp   LISTEN 0      4096         0.0.0.0:5001       0.0.0.0:*    users:(("docker-proxy",pid=133334,fd=4))
tcp   LISTEN 0      4096         0.0.0.0:81         0.0.0.0:*    users:(("docker-proxy",pid=132862,fd=4))
tcp   LISTEN 0      128             [::]:22            [::]:*    users:(("sshd",pid=621,fd=4))
tcp   LISTEN 0      4096            [::]:8088          [::]:*    users:(("docker-proxy",pid=132827,fd=4))
tcp   LISTEN 0      4096            [::]:443           [::]:*    users:(("docker-proxy",pid=132847,fd=4))
tcp   LISTEN 0      4096            [::]:5001          [::]:*    users:(("docker-proxy",pid=133339,fd=4))
tcp   LISTEN 0      4096               *:80               *:*    users:(("casaos",pid=591,fd=8))
tcp   LISTEN 0      4096            [::]:81            [::]:*    users:(("docker-proxy",pid=132867,fd=4))
tcp   LISTEN 0      4096               *:36147            *:*    users:(("casaos",pid=591,fd=9))`

I'm not certain where the discrepancy during installation is. The ports 5001, 81, 8088, are used by containers I run. Including a reverse proxy.

Just my two pence.
